docs/src/tutorials/tutorial_4.md

```diff
@@ -0,0 +1,304 @@
+# **Tutorial 4:** Do you know our Mascot? - Learn about transitions and subactions!
+
+You have learned a couple of cool features of Javis already. Now you're ready to finally meet our little mascot. Well actually you can't just see him, we have to create him first. :smile:
+
+Let's create a list of what we want first
+- a circular head
+- some hair
+- eyes
+- a nose
+- a moving mouth
+- he should say something
+
+This should show you the power of `SubAction` so we want each of those appear (fade in) one after another.
+
+```julia
+using Javis
+
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+function title(args...)
+    fontsize(20)
+    text("Our Mascot", Point(0, -200),
+        valign=:middle, halign=:center)
+end
+
+function face()
+    video = Video(500, 500)
+    javis(video, [
+        BackgroundAction(1:150, ground),
+        Action(title; subactions=[
+            SubAction(1:5, appear(:fade)),
+        ]),
+    ], pathname="jarvis.gif", framerate=15)
+end
+```
+
+The `ground` function should be familiar to you as well as the general structure of the `javis` function.
+
+A new part is the second `Action` which takes in a function and `subactions` as a keyword argument. 
+
+If you're wondering where the `frames` are specified:
+- You can leave the frames blank as here and then the frames from the previous action are used. 
+
+The `subactions` keyword uses a list of [`SubAction`](@ref) structs which are defined in a similar fashion with `frames` and a function but are in some sense simpler than the `Action`. 
```

Comment:
Could you explain here why we are using SubActions and why someone would want to use a SubAction?

Reply:
I think I explained it now in a few different ways and they are not that different from `Action`. If you think you don't understand how they work it would be great if you can clarify what you didn't understand or what users might not understand yet.

Reply:
I think this explanation is good now coupled with my suggestion in the prior conversation.